tests/components/homematic/test_notify.py

```diff
@@ -1,77 +1,65 @@
-            "homematic",
-            {"homematic": {"hosts": {"ccu2": {"host": "127.0.0.1"}}}},
+async def test_setup_without_optional(hass):
+    """Test valid configuration without optional."""
+    await async_setup_component(
+        hass,
+        "homematic",
+        {"homematic": {"hosts": {"ccu2": {"host": "127.0.0.1"}}}},
+    )
+    with assert_setup_component(1) as handle_config:
+        assert await async_setup_component(
+            hass,
+            "notify",
+            {
+                "notify": {
+                    "name": "test",
+                    "platform": "homematic",
+                    "address": "NEQXXXXXXX",
+                    "channel": 2,
+                    "param": "SUBMIT",
+                    "value": "1,1,108000,2",
+                }
+            },
         )
-        with assert_setup_component(1) as handle_config:
-            assert setup_component(
-                self.hass,
-                "notify",
-                {
-                    "notify": {
-                        "name": "test",
-                        "platform": "homematic",
-                        "address": "NEQXXXXXXX",
-                        "channel": 2,
-                        "param": "SUBMIT",
-                        "value": "1,1,108000,2",
-                        "interface": "my-interface",
-                    }
-                },
-            )
-        assert handle_config[notify_comp.DOMAIN]
+    assert handle_config[notify_comp.DOMAIN]
 
-    def test_setup_without_optional(self):
-        """Test valid configuration without optional."""
-        setup_component(
-            self.hass,
-            "homematic",
-            {"homematic": {"hosts": {"ccu2": {"host": "127.0.0.1"}}}},
-        )
-        with assert_setup_component(1) as handle_config:
-            assert setup_component(
-                self.hass,
-                "notify",
-                {
-                    "notify": {
-                        "name": "test",
-                        "platform": "homematic",
-                        "address": "NEQXXXXXXX",
-                        "channel": 2,
-                        "param": "SUBMIT",
-                        "value": "1,1,108000,2",
-                    }
-                },
-            )
-        assert handle_config[notify_comp.DOMAIN]
 
-    def test_bad_config(self):
-        """Test invalid configuration."""
-        config = {notify_comp.DOMAIN: {"name": "test", "platform": "homematic"}}
-        with assert_setup_component(0) as handle_config:
-            assert setup_component(self.hass, notify_comp.DOMAIN, config)
-        assert not handle_config[notify_comp.DOMAIN]
+async def test_bad_config(hass):
+    """Test invalid configuration."""
+    config = {notify_comp.DOMAIN: {"name": "test", "platform": "homematic"}}
+    with assert_setup_component(0) as handle_config:
+        assert await async_setup_component(hass, notify_comp.DOMAIN, config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)